include/cuco/static_set.cuh

```diff
@@ -0,0 +1,263 @@
+   *
+   * @param capacity The total number of slots in the set
+   * @param empty_key_sentinel The reserved key value for empty slots
+   * @param pred Key equality binary predicate
+   * @param probing_scheme Probing scheme
+   * @param alloc Allocator used for allocating device storage
+   * @param stream CUDA stream used to initialize the map
+   */
+  constexpr static_set(Extent capacity,
+                       empty_key<Key> empty_key_sentinel,
+                       KeyEqual pred                       = {},
+                       ProbingScheme const& probing_scheme = {},
+                       Allocator const& alloc              = {},
+                       cudaStream_t stream                 = nullptr);
+
+  /**
+   * @brief Inserts all keys in the range `[first, last)` and returns the number of successful
+   * insertions.
+   *
+   * @note This function synchonizes with the host thread. For asynchronous execution use
+   * `insert_async`.
+   *
+   * @tparam InputIt Device accessible random access input iterator where
+   * <tt>std::is_convertible<std::iterator_traits<InputIt>::value_type,
+   * static_set<K>::value_type></tt> is `true`
+   *
+   * @param first Beginning of the sequence of keys
+   * @param last End of the sequence of keys
+   * @param stream CUDA stream used for insert
+   *
+   * @return Number of successfully inserted keys
+   */
+  template <typename InputIt>
+  size_type insert(InputIt first, InputIt last, cudaStream_t stream = nullptr);
+
+  /**
+   * @brief Inserts all keys in the range `[first, last)`.
+   *
+   * @tparam InputIt Device accessible random access input iterator where
+   * <tt>std::is_convertible<std::iterator_traits<InputIt>::value_type,
+   * static_set<K>::value_type></tt> is `true`
+   *
+   * @param first Beginning of the sequence of keys
+   * @param last End of the sequence of keys
+   * @param stream CUDA stream used for insert
+   */
+  template <typename InputIt>
+  void insert_async(InputIt first, InputIt last, cudaStream_t stream = nullptr);
+
+  /**
+   * @brief Indicates whether the keys in the range `[first, last)` are contained in the set.
+   *
+   * @note This function synchonizes with the host thread. For asynchronous execution use
+   * `contains_async`.
+   *
+   * @tparam InputIt Device accessible input iterator
+   * @tparam OutputIt Device accessible output iterator assignable from `bool`
+   *
+   * @param first Beginning of the sequence of keys
+   * @param last End of the sequence of keys
+   * @param output_begin Beginning of the sequence of booleans for the presence of each key
+   * @param stream Stream used for executing the kernels
+   */
+  template <typename InputIt, typename OutputIt>
+  void contains(InputIt first,
+                InputIt last,
+                OutputIt output_begin,
+                cudaStream_t stream = nullptr) const;
+
+  /**
+   * @brief Indicates whether the keys in the range `[first, last)` are contained in the set.
```

Comment:
```suggestion
   * @brief Indicates whether the keys in the range `[first, last)` are contained in the set asynchronously.
```

Or `Asynchronously indicates whether the keys in the range [first, last) are contained in the set.`? which is better?